File: summary/ruby/10_cumulative_sum.rb
visitors = [62, 65, 41, 13, 20, 11, 18, 44, 53, 12, 18, 17, 14, 10, 39]
accumulates = Array.new(visitors.length + 1, 0)
(1..visitors.length).each do |i|
  accumulates[i] = accumulates[i - 1] + visitors[i - 1]
end
targets = [[2, 4], [4, 13]]
targets.each do |l, r|
  puts "#{l}日目から#{r}日目までの来場者数"
  puts accumulates[r] - accumulates[l - 1]
end

File: summary/rust/10_cumulative_sum.rs
fn main() {
    let visitors = vec![62, 65, 41, 13, 20, 11, 18, 44, 53, 12, 18, 17, 14, 10, 39];
    let mut accumulates = vec![0; visitors.len() + 1];
    for i in 1..=visitors.len() {
        accumulates[i] = accumulates[i - 1] + visitors[i - 1];
    }
    let targets = vec![(2, 4), (4, 13)];
    for (l, r) in targets.iter() {
        println!("{}日目から{}日目までの来場者数", l, r);
        println!("{}", accumulates[*r] - accumulates[l - 1]);
    }
}
